src/main/java/bio/terra/grammar/azure/SynapseVisitor.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.grammar.azure;
+
+import bio.terra.grammar.DatasetAwareVisitor;
+import bio.terra.grammar.SQLParser;
+import bio.terra.model.DatasetModel;
+import java.util.Map;
+
+public class SynapseVisitor extends DatasetAwareVisitor {
+
+  private static final int PRIME = 31;
+  private final String sourceDatasetDatasource;
+
+  public SynapseVisitor(Map<String, DatasetModel> datasetMap, String sourceDatasetDatasource) {
+    super(datasetMap);
+    this.sourceDatasetDatasource = sourceDatasetDatasource;
+  }
+
+  public String generateAlias(String tableName) {
+    int tableNameHash = tableName.hashCode();
+    // there's less of a chance of collision if we multiply the first value by an odd prime before
+    // we sum them
+    int hash = Math.abs(tableNameHash * PRIME);
+    return "alias" + hash;
```

Comment:
I don't see any summing here. But if you want to do this, it's the same as
```suggestion
    return "alias" + Objects.hash(tableName);
```

Reply:
Apologies, that comment was left over from the BigQuery version of this code. I've updated this in both places.
I think I still need the absolute value call if I don't want to potentially be negative, right? Synapse does not allow table names with dashes.

Reply:
I think you're right, Java doesn't require that hashcode be non-negative so `abs` is still required.